.github/scripts/update_generation_config.sh

```diff
@@ -1,12 +1,36 @@
 #!/bin/bash
 set -e
 # This script should be run at the root of the repository.
-# This script is used to update googleapis commit to latest in generation
+# This script is used to update parameters in generation
 # configuration at the time of running and create a pull request.
 
 # The following commands need to be installed before running the script:
 # 1. git
 # 2. gh
+# 3. jq
+
+# Utility functions
+function get_major_version() {
+    local key_word=$1
+    major=$(grep "${key_word}" "${generation_config}" | cut -d ':' -f 2 | xargs | cut -d '.' -f 1)
+    echo "${major}"
+}
+
+function get_latest_release_version() {
```

Comment:
Should this be "get_latest_released_version"?
Also, could you add a quick comment on these helper functions to explain what they do?

Reply:
Done.